packages/slice-machine/src/features/onboarding/useOnboarding.ts

```diff
@@ -0,0 +1,70 @@
+import { OnboardingStepId } from "@prismicio/editor-fields";
+import { updateData, useRequest } from "@prismicio/editor-support/Suspense";
+
+import { managerClient } from "@/managerClient";
+
+async function getOnboarding() {
+  try {
+    return await managerClient.prismicRepository.fetchOnboarding();
+  } catch (e) {
+    console.error("Error while trying to get onboarding", e);
+    return undefined;
+  }
+}
+
+export function useOnboarding() {
+  const onboarding = useRequest(getOnboarding, []);
+
+  async function toggleStep(stepId: OnboardingStepId) {
+    if (!onboarding) return [];
+
+    const previousSteps = onboarding.completedSteps;
+
+    const newSteps = previousSteps.includes(stepId)
+      ? previousSteps.filter((id) => id !== stepId)
+      : [...previousSteps, stepId];
+
+    updateData(getOnboarding, [], {
+      ...onboarding,
+      completedSteps: newSteps,
+    });
+
+    try {
+      const { completedSteps } =
+        await managerClient.prismicRepository.toggleOnboardingStep(stepId);
+
+      updateData(getOnboarding, [], {
+        ...onboarding,
+        completedSteps,
+      });
+
+      return completedSteps;
+    } catch (err) {
+      console.error("Error toggling step:", err);
```

Comment:
⚠️ What happens on the UI if you cannot complete a step? Here I don't see anything, but the user would not understand? Is that something you discussed with Design / Product? Should we have a toaster or something? It's part of the onboarding and I guess it's important to have the best flow even in case of errors.

Reply:
Asked Come about that and he decided to keep it super simple like it is now.

Reply:
Ok, for me it's not about "simple" but bug. A user that would get an error would not have something "simple" but  would experience a bug. The error is not a bug, not having feedback is for me. I would like we push product to ensure error feedback as for us it can be 2 min dev. 
This can even help engineering in case of problem because the user will report I saw an alert with this message.

I guess here it's "less" a problem than editor because dev often open their console but still.

Reply:
Change to show a small toast on error ✅